src/verifai/scenic_server.py

```diff
@@ -62,3 +73,160 @@ def _simulate(self, scene):
 
     def terminate(self):
         pass
+
+class DummySampler(VerifaiSampler):
+
+    def nextSample(self, feedback):
+        return self.last_sample, None
+
+@ray.remote
+class SampleSimulator():
+
+    def __init__(self, scenic_path, worker_num, monitor, options={}, use_carla=False,
+    scenario_params={}):
+        print(scenario_params)
+        scenario_params.update({
+            'port': 2000 + 2*worker_num
+        })
+        self.sampler = ScenicSampler.fromScenario(scenic_path, maxIterations=1, **scenario_params)
+        # reset self.sampler.scenario.externalSampler to dummy sampler
+        # that reads argument
+        self.worker_num = worker_num
+        self.sampler.scenario.externalSampler = DummySampler(self.sampler.scenario.externalParams,
+        self.sampler.scenario.params)
+        self.simulator = self.sampler.scenario.getSimulator()
+        self.monitor = monitor
+        extSampler = self.sampler.scenario.externalSampler
+        if extSampler is None:
+            self.rejectionFeedback = None
+        else:
+            self.rejectionFeedback = extSampler.rejectionFeedback
+        defaults = DotMap(maxSteps=None, verbosity=0, maxIterations=1)
+        defaults.update(options)
+        self.maxSteps = defaults.maxSteps
+        self.verbosity = defaults.verbosity
+        self.maxIterations = defaults.maxIterations
+
+    def get_sample(self, sample):
+        self.sampler.scenario.externalSampler.last_sample = sample
+        self.full_sample = self.sampler.nextSample(sample)
+
+    def simulate(self, sample):
+
+        '''
+        Need to generate scene from sample here.
+        '''
+        t0 = time.time()
+        self.sampler.scenario.externalSampler.last_sample = sample
+        scene = self.sampler.lastScene
+        startTime = time.time()
+        if self.verbosity >= 1:
+            print('  Beginning simulation...')
+        try:
+            result = self.simulator.simulate(scene,
+                maxSteps=self.maxSteps, verbosity=self.verbosity,
+                maxIterations=self.maxIterations)
+            result.worker_num = self.worker_num
+        except SimulationCreationError as e:
+            if self.verbosity >= 1:
+                print(f'  Failed to create simulation: {e}')
+            self.lastValue = self.rejectionFeedback
+            return self.worker_num, self.full_sample, self.lastValue
+        except RuntimeError as e:
+            print(f'Runtime error during simulation: {e}')
+            print('Waiting 1 minute before continuing...')
+            time.sleep(60)
+            return self.worker_num, self.full_sample, self.lastValue
```

Comment:
What is this code supposed to handle? Catching all RuntimeErrors is pretty broad.